common.mk

```diff
@@ -142,15 +141,34 @@ FIRTOOL_TARGETS = \
 	$(FIRTOOL_SMEMS_CONF) \
 	$(FIRTOOL_TOP_SMEMS_JSON) \
 	$(FIRTOOL_TOP_HRCHY_JSON) \
-	$(FIRTOOL_MODEL_MOD_HRCHY_JSON) \
+	$(FIRTOOL_MODEL_HRCHY_JSON) \
 	$(FIRTOOL_MODEL_SMEMS_JSON) \
 	$(FIRTOOL_FILELIST) \
 	$(FIRTOOL_BB_MODS_FILELIST)
 
+ifeq (,$(ENABLE_CUSTOM_FIRRTL_PASS))
+	REPL_SEQ_MEM = none
+	TOP_TARGETS = none
+	HARNESS_TARGETS = none
+	TRANSFORMS = barstools.tapeout.transforms.GenerateTop
+else
+	REPL_SEQ_MEM = --infer-rw --repl-seq-mem -c:$(MODEL):-o:$(TOP_SMEMS_CONF)
+	TOP_TARGETS = $(TOP_SMEMS_CONF)
+	HARNESS_CONF_FLAGS = -thconf $(HARNESS_SMEMS_CONF)
+	HARNESS_TARGETS = $(HARNESS_SMEMS_CONF)
+	TRANSFORMS = barstools.tapeout.transforms.GenerateTopAndHarness
```

Comment:
Technically this is the SFC name that holds the xforms. Maybe rename it to something like `SFC_MAIN`?